mlflow/tracing/provider.py

```diff
@@ -114,6 +114,34 @@ def _force_set_otel_tracer_provider(tracer_provider):
     trace.set_tracer_provider(tracer_provider)
 
 
+def _catch_exception(msg: str, default_return=None):
```

Comment:
Trying to precisely catch exceptions from finally clause requires dirty nesting of try-catch. So I ended up making `enable()` / `disable()` / `_is_enabled()` exception-free, and use this decorator as forcing function to ensure that. It may still not be the best option, so please let me know if you have other thoughts on this @harupy @serena-ruan　🙌

Reply:
My concern is that for `trace_disabled` if anything fails we print warnings 🤔 We might want to use debug for it.
Can we update trace_disabled:
```
@functools.wraps(f)
    def wrapper(*args, **kwargs):
        catch_exception(disable)()
        try:
            return f(*args, **kwargs)
        finally:
            catch_exception(enable)()
        else:
            return f(*args, **kwargs)

    return wrapper
```
and `catch_exception` accepts a function as input and do a similar thing but uses `logger.debug`?

Reply:
Discussed offline, we'll address this in 2.14.0 given that these API is not publicly used until then.